grants_tagger/__main__.py

```diff
@@ -205,49 +212,59 @@ def bioasq_mesh(
         cfg.read(config)
 
         input_path = cfg["preprocess"]["input"]
-        output_path = cfg["preprocess"]["output"]
+        train_output_path = cfg["preprocess"]["output"]
         mesh_tags_path = cfg["filter_mesh"].get("mesh_tags_path")
         test_split = cfg["preprocess"].getfloat("test_split")
 
-    # TODO: Refactor with preprocess_mesh
-    if test_split:
-        data_dir, data_name = os.path.split(output_path)
-        train_output_path = os.path.join(data_dir, "train_" + data_name)
-        test_output_path = os.path.join(data_dir, "test_" + data_name)
+    if verify_if_paths_exist(
+        [
+            input_path,
+            train_output_path,
+            label_binarizer_path,
+            test_output_path,
+        ]
+    ):
+        return
 
-        if os.path.exists(train_output_path) and os.path.exists(test_output_path):
-            print(
-                f"{train_output_path} and {test_output_path} exists. Remove them if you want to rerun."
-            )
+    temporary_output_path = train_output_path + ".tmp"
```

Comment:
why is that needed? to reduce space needed?

Reply:
yes, that's what I thought. Cause first you do the preprocessing and you need a file to output that to. And then you create the train/test from it.

Reply:
and also this file is 20GB, good idea. if you were doing it to pass it in the label binarizer I would say that you can safely pass just the train there. but in the case of reducing size makes sense